code/controllers/ContentController.php

```diff
@@ -87,42 +88,58 @@ public function Page($link) {
 
 	public function init() {
 		parent::init();
-		
+
+		$recordExists = ($this->dataRecord && $this->dataRecord->exists());
```

Comment:
On the security page, we create a new `Page` (not `SiteTree`) record, and don't save it to the database. This will cause $recordExists to be false.

Have you checked that this doesn't interfere with any actions on the Security controller? I'm not saying that there are regressions, but there's a chance for them to occur here.